e2e/test_operator_ha/operator_ha_test.go

```diff
@@ -207,4 +210,93 @@ var _ = Describe("Operator HA tests", Label("e2e", "pr"), func() {
+
+	// Right now this test case doesn't work as there is no way to recover when the majority of coordinators are failing.
+	PWhen("all Pods in the primary and primary satellite failing", func() {
+		BeforeEach(func() {
+			primary := fdbCluster.GetPrimary()
+			primary.SetCrashLoopContainers([]fdbv1beta2.CrashLoopContainerObject{
+				{
+					ContainerName: fdbv1beta2.MainContainerName,
+					Targets:       []fdbv1beta2.ProcessGroupID{"*"},
+				},
+			}, false)
+
+			primarySatellite := fdbCluster.GetPrimarySatellite()
+			primarySatellite.SetCrashLoopContainers([]fdbv1beta2.CrashLoopContainerObject{
+				{
+					ContainerName: fdbv1beta2.MainContainerName,
+					Targets:       []fdbv1beta2.ProcessGroupID{"*"},
+				},
+			}, false)
+
+			primaryPods := primary.GetPods()
+			for _, pod := range primaryPods.Items {
+				factory.DeletePod(&pod)
+			}
+
+			primarySatellitePods := primarySatellite.GetPods()
+			for _, pod := range primarySatellitePods.Items {
+				factory.DeletePod(&pod)
+			}
+
+			remote := fdbCluster.GetRemote()
+			Eventually(func(g Gomega) bool {
+				out, _, err := remote.RunFdbCliCommandInOperatorWithoutRetry("status json", false, 30)
+				log.Println("Fetched status", out, err)
+				g.Expect(err).NotTo(HaveOccurred())
+
+				if strings.HasPrefix(out, "\r\nWARNING") {
+					out = strings.TrimPrefix(
+						out,
+						"\r\nWARNING: Long delay (Ctrl-C to interrupt)\r\n",
+					)
+				}
+
+				status := &fdbv1beta2.FoundationDBStatus{}
+				err = json.Unmarshal([]byte(out), status)
+				g.Expect(err).NotTo(HaveOccurred())
+				return status.Client.DatabaseStatus.Available
+			}).WithTimeout(5 * time.Minute).WithPolling(1 * time.Second).MustPassRepeatedly(4).Should(BeFalse())
+		})
+
+		It("should change the coordinators", func() {
+			newCoordinators := make([]fdbv1beta2.ProcessAddress, 0, 5)
+			remote := fdbCluster.GetRemote()
+			remotePods := remote.GetLogPods()
+
+			for _, pod := range remotePods.Items {
+				if len(newCoordinators) >= 3 {
+					break
+				}
+				addr, err := fdbv1beta2.ParseProcessAddress(fmt.Sprintf("%s:4500:tls", pod.Status.PodIP))
+				Expect(err).NotTo(HaveOccurred())
+				newCoordinators = append(newCoordinators, addr)
+			}
+
+			remoteSatellite := fdbCluster.GetRemoteSatellite()
+			remoteSatellitePods := remoteSatellite.GetLogPods()
+
+			for _, pod := range remoteSatellitePods.Items {
+				if len(newCoordinators) >= 5 {
+					break
+				}
+				addr, err := fdbv1beta2.ParseProcessAddress(fmt.Sprintf("%s:4500:tls", pod.Status.PodIP))
+				Expect(err).NotTo(HaveOccurred())
+				newCoordinators = append(newCoordinators, addr)
+			}
+
+			coordinatorCmd := fmt.Sprintf(
+				"coordinators %s",
+				fdbv1beta2.ProcessAddressesString(newCoordinators, " "),
+			)
+			log.Println("Command:", coordinatorCmd)
+
+			stdout, stderr, err := remote.RunFdbCliCommandInOperatorWithoutRetry(coordinatorCmd, true, 40)
+			log.Println("stdout:\n", stdout, "\nstderr:\n", stderr)
+			Expect(err).NotTo(HaveOccurred())
```

Comment:
In the current state, we'd expect an error here because this is not currently possible without a majority of current coordinators? https://apple.github.io/foundationdb/configuration.html

Reply:
Correct